src/api/services/reportingPeriodService.ts

```diff
@@ -18,18 +18,23 @@ class ReportingPeriodService {
       year: string
     }
   ) {
-    const reportingPeriod = await prisma.reportingPeriod.findFirst({
-      where: {
-        companyId: company.wikidataId,
-        year,
-      },
-    })
-
     return prisma.reportingPeriod.upsert({
       where: {
-        id: reportingPeriod?.id ?? '',
+        companyId_year: {
+          companyId: company.wikidataId,
+          year,
+        },
+      },
+      update: {
+        startDate,
+        endDate,
```

Comment:
This works in most cases. However, I think the reason we kept the `update` block empty was that we most of the time wanted to `create` the reporting period and/or return it, but not necessarily update it at the same time.

We could try this approach and hopefully it works going forward. But if you get future problems with garbo changing the reportingPeriod `startDate` and/or `endDate` (potentially causing conflicts if there are multiple reports for the same year), then consider separating the endpoints for creating and updating reportingPeriods.